### ISSUE TITLE
IPAddr::InvalidAddressError: invalid address when running bundle in Ruby 2.4.0

### ISSUE THREAD

**Issue by stefankroes** [bundler/bundler#5300]
- What did you do?

  I ran the command `/Users/stefan/.rvm/gems/ruby-2.4.0/bin/bundle `

- What did you expect to happen?

  I expected Bundler to install the gems required by my Gemfile.lock

- What happened instead?

  Instead, what happened was I got this error. I investigated a bit and it happens at a point where `self.host` evaluates to somehost.somelocaldomain.net and so `IPAddr#include?(self.host)` raises this error. I have one gem in my Gemfile with this source:

```
source 'http://somehost.somelocaldomain.net' do
  gem 'yyy'
end
```

I see that this area in the code has to do with proxy settings and I have those as well: HTTP_PROXY, HTTPS_PROXY, NO_PROXY=localhost,127.0.0.1,somelocaldomain.net, ... some IPs

Error details

    IPAddr::InvalidAddressError: invalid address
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/ipaddr.rb:563:in `in6_addr'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/ipaddr.rb:500:in `initialize'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/ipaddr.rb:518:in `new'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/ipaddr.rb:518:in `coerce_other'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/ipaddr.rb:174:in `include?'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/uri/generic.rb:1542:in `block in find_proxy'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/uri/generic.rb:1530:in `scan'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/uri/generic.rb:1530:in `find_proxy'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/net/http.rb:1061:in `proxy_uri'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/net/http.rb:1048:in `proxy?'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/net/http.rb:893:in `connect'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/net/http.rb:887:in `do_start'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/net/http.rb:882:in `start'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/net/http/persistent.rb:700:in `start'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/net/http/persistent.rb:631:in `connection_for'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/net/http/persistent.rb:994:in `request'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/fetcher/downloader.rb:48:in `request'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/fetcher/downloader.rb:16:in `fetch'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/fetcher/compact_index.rb:115:in `call'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/compact_index_client/lib/compact_index_client/updater.rb:42:in `block in update'
      /Users/stefan/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/tmpdir.rb:89:in `mktmpdir'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/compact_index_client/lib/compact_index_client/updater.rb:29:in `update'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/compact_index_client/lib/compact_index_client.rb:65:in `update'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/compact_index_client/lib/compact_index_client.rb:56:in `update_and_parse_checksums!'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/fetcher/compact_index.rb:67:in `available?'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/fetcher/compact_index.rb:15:in `call'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/fetcher/compact_index.rb:15:in `block in compact_index_request'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/fetcher.rb:157:in `use_api'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/source/rubygems.rb:332:in `block in api_fetchers'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/source/rubygems.rb:332:in `select'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/source/rubygems.rb:332:in `api_fetchers'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/source/rubygems.rb:337:in `block in remote_specs'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/index.rb:10:in `build'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/source/rubygems.rb:336:in `remote_specs'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/source/rubygems.rb:83:in `specs'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/lazy_specification.rb:61:in `__materialize__'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/spec_set.rb:91:in `block in materialize'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/spec_set.rb:88:in `map!'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/spec_set.rb:88:in `materialize'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/definition.rb:174:in `specs'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/definition.rb:162:in `resolve_remotely!'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/installer.rb:225:in `resolve_if_need'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/installer.rb:78:in `run'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/installer.rb:24:in `install'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/cli/install.rb:71:in `run'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/cli.rb:189:in `install'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/cli.rb:20:in `dispatch'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/cli.rb:11:in `start'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/exe/bundle:34:in `block in <top (required)>'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/lib/bundler/friendly_errors.rb:100:in `with_friendly_errors'
      /Users/stefan/.rvm/gems/ruby-2.4.0/gems/bundler-1.13.7/exe/bundle:26:in `<top (required)>'
      /Users/stefan/.rvm/gems/ruby-2.4.0/bin/bundle:22:in `load'
      /Users/stefan/.rvm/gems/ruby-2.4.0/bin/bundle:22:in `<main>'
      /Users/stefan/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `eval'
      /Users/stefan/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `<main>'

Environment
      
  Bundler   1.13.7
  Rubygems  2.6.8
  Ruby      2.4.0p0 (2016-12-24 revision 57164) [x86_64-darwin16]
  GEM_HOME  /Users/stefan/.rvm/gems/ruby-2.4.0
  GEM_PATH  /Users/stefan/.rvm/gems/ruby-2.4.0:/Users/stefan/.rvm/gems/ruby-2.4.0@global
  RVM       1.28.0 (latest)
  Git       2.10.1 (Apple Git-78)
  rubygems-bundler (1.4.4)

**Comment by stefankroes** [comment]
Changing ...

```
              return nil if
                begin
                  IPAddr.new(host).include?(self.host)
                rescue IPAddr::InvalidAddressError
                  next
                end
```

... to ...

```
              return nil if
                begin
                  IPAddr.new(host).include?(self.host)
                rescue IPAddr::InvalidAddressError
                  next
                end
```

... seems to resolve the issues for me but I'm unsure if this is the right solution.

**Comment by segiddins** [comment]
This looks to me to be a bug in ruby rather than in bundler?

**Comment by stefankroes** [comment]
You are probably right, sorry :-(

**Comment by roberts1000** [comment]
I just hit this too.  Looks like it might be fixed when ruby 2.4.1 comes out by https://github.com/ruby/ruby/pull/1513
